csrc/runtime/fusion_kernel_runtime.cpp

```diff
@@ -445,10 +449,39 @@ void FusionKernelRuntime::compileFusionParallel(KernelArgumentHolder args) {
             std::vector<Val*>{out_clone});
         hic->pushBackTopLevelExprs(launch_kernel);
       } else {
-        // push back segment's exprs into the container as top level expressions
-        for (auto* expr : group_to_run->exprs()) {
-          auto cloned_expr = ir_cloner.clone(expr);
-          hic->pushBackTopLevelExprs(cloned_expr);
+        const bool is_resharding = std::any_of(
+            group_to_run->exprs().begin(), group_to_run->exprs().end(), [](auto expr) {
+              return isResharding(expr);
+            });
+        if (is_resharding) {
+          NVF_ERROR(
+              group_to_run->exprs().size() == 1,
+              "Communication segments must contain only one Expr");
+          HostIrLower lower;
+          for (auto* expr :
+              lower.lower(ir_cloner.clone(group_to_run->exprs().at(0)))) {
+            // Allocate the recv buffers of communications
```

Comment:
Can this change go to a different PR? https://google.github.io/eng-practices/review/developer/small-cls.html

Reply:
This part is copied from https://github.com/NVIDIA/Fuser/blob/f58617e8edfa13ad5e040046f3549d94402b3a45/csrc/host_ir/lower.cpp#L646 actually. Is it worth another PR?